src/dbnode/encoding/m3tsz/timestamp_iterator.go

```diff
@@ -269,14 +267,13 @@ func (it *TimestampIterator) readDeltaOfDelta(
 	if err != nil {
 		return 0, err
 	}
-
 	dod := encoding.SignExtend(dodBits, numValueBits)
 	timeUnit, err := it.TimeUnit.Value()
 	if err != nil {
 		return 0, nil
 	}
 
-	return xtime.FromNormalizedDuration(dod, timeUnit), nil
+	return xtime.ToUnixNanoDuration(xtime.FromNormalizedDuration(dod, timeUnit)), nil
```

Comment:
Is this necessary? Seems like you're converting to a duration then back to unix nanos

Reply:
Similarly here can you just keep the return of this method `readDeltaOfDelta` as `time.Duration`?